docs/type-long-integer.md

```diff
@@ -0,0 +1,96 @@
+# Long Integer Type
+
+## Summary
+
+A builtin type to represent 64-bit integers.
+
+## Motivation
+
+Current methods of representing 64-bit integers in Luau are done with high-low pairs or by splitting them among vectors which leads to poor ergonomics and may not be sufficient for cases where performance is important.
+
+In the case of a high-low pair implementation each individual number takes 16 bytes (or 24) and needs to be be handled together which can be confusing.
+
+In the case of vector implementations, ergonomics are improved by storing the entire integer in one value however you have the same issues of implementation complexity and lack of integration with the type system.
```

Comment:
Custom userdata can also integrate into the type system, no one is forced to use the trick with worse ergonomics.